src/Microsoft.IIS.WebManager/app/common/component-loader.ts

```diff
@@ -1,11 +1,155 @@
-declare var System: any;
+import { Type, Compiler, ModuleWithComponentFactories } from '@angular/core'
+import { WebSiteListComponentName,
+    AppPoolComponentName,
+    WebFilesComponentName,
+    WebAppListComponentName,
+    VdirListComponentName,
+    AuthenticationComponentName,
+    AuthorizationComponentName,
+    CertificatesComponentName,
+    CentralCertificateComponentName,
+    DefaultDocumentsComponentName,
+    DirectoryBrowsingComponentName,
+    IpRestrictionsComponentName,
+    LoggingComponentName,
+    MimeMapsComponentName,
+    MonitoringComponentName,
+    ModulesComponentName,
+    CompressionComponentName,
+    RequestFilteringComponentName,
+    HttpResponseHeadersComponentName,
+    RequestTracingComponentName,
+    StaticContentComponentName,
+    UrlRewriteComponentName,
+    ComponentReference,
+    FilesComponentName} from '../main/settings'
 
-export class ComponentLoader {   
 
-    public static LoadAsync(name, path) {
-        return System.import(path)
-            .then(m => {
-                return m[name]
-            });
+export class ComponentLoader {
+    public static async LoadAsync(compiler: Compiler, component: ComponentReference): Promise<ModuleWithComponentFactories<{}>> {
+        // cannot assign module type as property of ComponentReference to avoid circular reference
+        var moduleType : Promise<Type<{}>> = null
+        switch (component.component_name) {
+            case WebSiteListComponentName:
```

Comment:
Angular CLI statically analyze which code do bring in during compile time. I cannot pass a expression to System.Import function anymore. The parameter needs to be a string literal for the code to work